nixos/modules/services/desktops/xkb.nix

```diff
@@ -41,19 +148,17 @@ in
         X keyboard variant.
       '';
     };
-
-    dir = lib.mkOption {
-      type = lib.types.path;
-      default = "${pkgs.xkeyboard_config}/etc/X11/xkb";
-      defaultText = lib.literalExpression ''"''${pkgs.xkeyboard_config}/etc/X11/xkb"'';
-      description = ''
-        Path used for -xkbdir xserver parameter.
-      '';
-    };
   };
 
   config = lib.mkIf cfg.enable {
-    environment.etc."X11/xkb".source = config.services.xkb.dir;
+    environment = {
+      etc."X11/xkb".source = config.services.xkb.dir;
+      # runtime override supported by multiple libraries e. g. libxkbcommon
+      # https://xkbcommon.org/doc/current/group__include-path.html
+      sessionVariables.XKB_CONFIG_ROOT = config.services.xkb.dir;
```

Comment:
Regarding the commit message

> environment.sessionVariables.XKB_CONFIG_ROOT is not always set to make
> things more consistent.

Did you mean “_now_ always set” as it was not always set before either?